pkg/eniconfig/eniconfig.go

```diff
@@ -131,8 +131,10 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 
 			h.controller.eniLock.Lock()
 			defer h.controller.eniLock.Unlock()
-			h.controller.myENI = val
-			log.Infof("Setting myENI to: %s", val)
+			if h.controller.myENI != val {
```

Comment:
If you're checking this variable, you might as well move the lock inside the if-statement as well.

Reply:
Ah, nice catch. Fixed.